src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractMoveCommand.cs

```diff
@@ -0,0 +1,116 @@
+            {
+                return await GetCommandStatusResult.Handled(commandText, CommandStatus.Ninched).ConfigureAwait(true);
+            }
+
+            return await GetCommandStatusResult.Handled(commandText, CommandStatus.Enabled).ConfigureAwait(true);
+        }
+
+        protected override async Task<bool> TryHandleCommandAsync(IProjectTree node, bool focused, long commandExecuteOptions, IntPtr variantArgIn, IntPtr variantArgOut)
+        {
+            var didMove = await Move(_configuredProject, node).ConfigureAwait(true);
+
+            if (didMove)
+            {
+                // Wait for updating to finish before re-selecting the node that moved.
+                // We need to re-select the node after it is moved in order to continuously move the node using hotkeys.
+                await _projectTree.TreeService.PublishLatestTreeAsync(waitForFileSystemUpdates: true).ConfigureAwait(true);
+
+                if (_configuredProject.UnconfiguredProject.Services.HostObject is IVsUIHierarchy hierarchy)
+                {
+                    // The node moved will retain its identity.
+                    var itemId = (uint)node.Identity.ToInt32();
+                    var window = GetUIHierarchyWindow(_serviceProvider, Guid.Parse(ManagedProjectSystemPackage.SolutionExplorerGuid));
+
+                    window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_SelectItem);
+                    
+                    // If we moved a folder, collapse it.
+                    if (node.IsFolder)
+                    {
+                        window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_CollapseFolder);
+                    }
+                }
+            }
+
+            return didMove;
+        }
+
+        /// <summary>
+        /// Get reference to IVsUIHierarchyWindow interface from guid persistence slot.
+        /// </summary>
+        /// <param name="serviceProvider">The service provider.</param>
+        /// <param name="persistenceSlot">Unique identifier for a tool window created using IVsUIShell::CreateToolWindow.
+        /// The caller of this method can use predefined identifiers that map to tool windows if those tool windows
+        /// are known to the caller. </param>
+        /// <returns>A reference to an IVsUIHierarchyWindow interface, or <c>null</c> if the window isn't available, such as command line mode.</returns>
+        private static IVsUIHierarchyWindow GetUIHierarchyWindow(IServiceProvider serviceProvider, Guid persistenceSlot)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            IVsUIShell shell = serviceProvider.GetService(typeof(SVsUIShell)) as IVsUIShell;
+
+            if (shell == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            object pvar = null;
+            IVsWindowFrame frame = null;
```

Comment:
fame declaration can be inlined `shell.FindToolWindow(0, ref persistenceSlot, out var frame)`

Reply:
Yup.